pwnlib/heap/glmalloc/malloc_state/malloc_state/parser.py

```diff
@@ -0,0 +1,260 @@
+            malloc_state = self.parse_from_address(malloc_state_address)
+            malloc_states.append(malloc_state)
+            malloc_state_address = malloc_state.next
+
+        return malloc_states
+
+    def parse_from_address(self, address):
+        """Parses a malloc_state struct from the process memory
+
+        Returns:
+            MallocState
+        """
+        raw_malloc_state = self._process_informer.read_memory(
+            address,
+            self.raw_malloc_state_size
+        )
+        return self.parse_from_raw(address, raw_malloc_state)
+
+    def parse_from_raw(self, address, raw_malloc_state):
+        """Parses a binary malloc_state struct
+
+        Args:
+            address (int): address of the data, useful as metadata for bins arrays
+            raw_malloc_state (bytearray): malloc_state in binary form
+
+        Returns:
+            MallocState
+        """
+        malloc_state_collection = self._malloc_state_struct_definition.parse(
+            raw_malloc_state
+        )
+
+        mutex = malloc_state_collection.mutex
+        flags = malloc_state_collection.flags
+        try:
+            have_fastchunks = malloc_state_collection.have_fastchunks
+        except AttributeError:
+            have_fastchunks = None
+
+        fastbinsY_address = address + \
+            self._malloc_state_struct_definition.fastbinsY_offset
+        fastbinsY = self._fastbinsy_parser.parse_from_collection(
+            fastbinsY_address,
+            malloc_state_collection.fastbinsY
+        )
+
+        top = malloc_state_collection.top
+        last_remainder = malloc_state_collection.last_remainder
+
+        bins_address = address + \
+            self._malloc_state_struct_definition.bins_offset
+        bins = self._bins_parser.parse_from_collection(
+            bins_address,
+            malloc_state_collection.bins
+        )
+
+        binmap = list(malloc_state_collection.binmap)
+        next_ = malloc_state_collection.next
+        next_free = malloc_state_collection.next_free
+        try:
+            attached_threads = malloc_state_collection.attached_threads
+        except AttributeError:
+            attached_threads = None
+
+        system_mem = malloc_state_collection.system_mem
+        max_system_mem = malloc_state_collection.max_system_mem
+
+        return MallocState(
+            address,
+            mutex,
+            flags,
+            have_fastchunks,
+            fastbinsY,
+            top,
+            last_remainder,
+            bins,
+            binmap,
+            next_,
+            next_free,
+            attached_threads,
+            system_mem,
+            max_system_mem
+        )
+
+    def _calculate_raw_malloc_state_size(self):
+        return self._malloc_state_struct_definition.size
+
+
+class MallocStateStructSelector:
+
+    def __init__(self, libc_version, pointer_size):
+        self._pointer_size = pointer_size
+        if pointer_size == 8:
```

Comment:
pointer_size should be deduced from `context.bytes`, not passed in as a parameter

Reply:
I disagree, this is an internal function and I think is a bad practice to rely on a global variable. Moreover, the pointer_size is based on the pointer size of the process/corefile, that can differ from the machine bytes represented by `context.bytes`.